prow/plugins/golint/golint.go

```diff
@@ -224,14 +255,20 @@ func handle(minimumConfidence float64, ghc githubClient, gc *git.Client, log *lo
 	}
 	log.WithField("duration", time.Since(finishClone)).Info("Linted.")
 
-	oldComments, err := ghc.ListPullRequestComments(org, repo, e.Number)
-	if err != nil {
-		return err
+	nps := problems
+	if len(problems) > 0 {
+		oldComments, err := ghc.ListPullRequestComments(org, repo, e.Number)
+		if err != nil {
+			return err
+		}
+		nps = newProblems(oldComments, problems)
 	}
-	nps := newProblems(oldComments, problems)
 
 	// Make the list of comments.
 	var comments []github.DraftReviewComment
+	if lintErrorComments != nil {
+		comments = lintErrorComments
+	}
```

Comment:
This conditional doesn't change the behavior. `comments := lintErrorComments` is sufficient.